torch/_functorch/aot_autograd.py

```diff
@@ -3114,11 +3133,19 @@ def forward(ctx, *deduped_flat_tensor_args):
             # invariant: intermediate bases always require gradients, so we don't have to
             # consider marking them as non-differentiable.
             raw_returns_not_including_intermediate_bases = raw_returns[:num_mutated_inputs + num_outputs]
+
+            raw_returns_meta = (
+                [x for x in CompiledFunction.metadata.input_info if x.mutates_data or x.mutates_metadata]
+                + CompiledFunction.metadata.output_info
+            )
+
+            for (i, x) in enumerate(raw_returns_not_including_intermediate_bases):
```

Comment:
Maybe assert that `len(raw_returns_not_including_intermediate_bases) == len(raw_returns_meta)` too?